gobblin-core/src/main/java/gobblin/writer/ParquetDataWriterBuilder.java

```
@@ -0,0 +1,90 @@
+      Path stagingFile;
+
+      String filePath = properties
+              .getProp(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_PATH, branch));
+      // Add the writer ID to the file name so each writer writes to a different
+      // file of the same file group defined by the given file name
+      String fileName = String.format("%s.%s.%s", properties
+              .getProp(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_NAME, branch),
+                      "part"), writerId, "tmp");
+      String stagingDir =
+              properties.getProp(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_STAGING_DIR, branch)) +
+                      Path.SEPARATOR + filePath;
+
+      stagingFile = new Path(stagingDir, fileName);
+      FileSystem fs = FileSystem.get(URI.create(uri), conf);
+      // Deleting the staging file if it already exists, which can happen if the
+      // task failed and the staging file didn't get cleaned up for some reason.
+      // Deleting the staging file prevents the task retry from being blocked.
+      if (fs.exists(stagingFile)) {
+        LOG.warn(String.format("Task staging file %s already exists, deleting it", stagingFile));
+        fs.delete(stagingFile, false);
+      }
+
+      Class<? extends Message> cls = null;
+      try {
+        cls = (Class<? extends Message>) Class.forName(this.schema);
+      } catch (Exception e) {
+        throw new IOException(e);
+      }
+      // todo set compression?
+      protoWriter = new ParquetWriter<Message>(stagingFile, new ProtoWriteSupport(cls), CompressionCodecName.UNCOMPRESSED, BLOCK_SIZE, PAGE_SIZE, PAGE_SIZE, true, true, ParquetWriter.DEFAULT_WRITER_VERSION, conf);
+      return new ParquetDataWriter(properties, this.writerId, this.branch, protoWriter, stagingFile, conf);
+    } else if (recordType.equals(ConfigurationKeys.PARQUET_RECORD_AVRO)) {
+      return null; // not yet implemented
```

Should we raise an exception here rather than failing silently?